def with_catch
  yield
rescue Exception => e
  puts
  puts "//////////////////////////////////////////////////////////////////////////////////////////////////////////////"
  puts; puts e.message; puts
  puts e.backtrace
  puts
  puts "//////////////////////////////////////////////////////////////////////////////////////////////////////////////"
  puts
  status 500
end